components/server/src/main/java/com/hotels/styx/server/netty/connectors/HttpResponseWriter.java

```diff
@@ -157,7 +158,7 @@ private void onWriteEmptyLastChunkOutcome(ChannelFuture writeOp) {
             return future;
         } catch (Throwable cause) {
             LOGGER.warn("Failed to convert response headers. response={}, Cause={}", new Object[]{response, cause});
-            response.releaseContentBuffers();
+            toObservable(response.body()).forEach(it -> Buffers.toByteBuf(it).release());
```

Comment:
This could be a stupid comment: do we need to expose a method to release the content buffers to plugins?  In that case, it could also be used here.

Reply:
This is in fact an excellent comment. In my opinion, if possible, we should not expose reference counts outside of `Buffer` at all.

However we need to ensure this indeed is the case, and one place we need to revisit is `ByteStream.aggregate` method. It returns an aggregated `Buffer` but the end-users have no way of releasing it yet.

I would like to implement these improvements in the next round though.